### ISSUE TITLE
Cached item not found during github action

### ISSUE THREAD

**Issue by jfcrenshaw** [showyourwork/showyourwork#102]
I have two trained models cached in a zenodo sandbox draft:
<img width="1157" alt="image" src="https://user-images.githubusercontent.com/41785729/170741271-93b950fe-d27f-4a6c-b420-049346baebd6.png">

Both are used in plotting scripts to make figures for [my paper](https://github.com/jfcrenshaw/pzflow-paper).

The two moons flow is found and the plot is made, but the main galaxy flow is not. During the [github action](https://github.com/jfcrenshaw/pzflow-paper/actions/runs/2397451052), I get the error

> ```
> Running user rule train_main_galaxy_flow...
> Searching remote file cache: src/data/main_galaxy_flow.pzflow.pkl...
> Required version of file not found in cache: src/data/main_galaxy_flow.pzflow.pkl.
> Cache for src/data/main_galaxy_flow.pzflow.pkl not found, and `run_cache_rules_on_ci` is set to `False`.
> ```

The corresponding rule in my [`Snakefile`](https://github.com/jfcrenshaw/pzflow-paper/blob/main/Snakefile) is
```
rule train_main_galaxy_flow:
    output:
        "src/data/main_galaxy_flow.pzflow.pkl"
    cache:
        True
    script:
        "src/scripts/train_main_galaxy_flow.py"
```

And the main galaxy flow is listed as a dependency in my [`showyourwork.yml`](https://github.com/jfcrenshaw/pzflow-paper/blob/main/showyourwork.yml):
```
dependencies:
  src/scripts/plot_two_moons.py:
    - src/data/two_moons_flow.pzflow.pkl
  src/scripts/plot_main_galaxy_corner.py:
    - src/data/main_galaxy_flow.pzflow.pkl
```

Everything runs okay on my laptop, and I've checked what I can think of, so I don't know what's going wrong. Especially since it is able to restore the two moons flow from the Zenodo Sandbox cache.

**Comment by rodluger** [comment]
The workflow only restores a file from the cache if the hash of the rule +
all dependencies is an exact match. Are you sure you committed and pushed
all your changes to the remote? Perhaps the action triggered before the
local build finished and updated the cache? If none of that applies, try
doing a make clean and rebuilding locally. When it's done, trigger the
GitHub action. If that fails let me know and I'll investigate!

**Comment by jfcrenshaw** [comment]
I tried cleaning and re-running everything, but still no luck. I also tried deleting the zenodo sandbox cache and creating a new one, but still no luck - now nothing is being synced. I tried from two different computers using two different sandbox tokens. Any ideas?

Also, not sure if this is related, but when I run `showyourwork clean`, I get the following error:
```
❯ showyourwork clean
Traceback (most recent call last):
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/bin/showyourwork", line 8, in <module>
    sys.exit(entry_point())
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/showyourwork/cli/main.py", line 266, in clean
    commands.clean(force, deep)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/site-packages/showyourwork/cli/commands/clean.py", line 18, in clean
    shutil.rmtree(paths.user().temp)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/shutil.py", line 732, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/shutil.py", line 665, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/shutil.py", line 688, in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
  File "/astro/store/epyc/users/jfc20/local/mambaforge/envs/pzflow-paper/lib/python3.9/shutil.py", line 686, in _rmtree_safe_fd
    os.unlink(entry.name, dir_fd=topfd)
OSError: [Errno 16] Device or resource busy: '.nfs00000000817d08230000020e'
```
It still deletes all of the intermediate files in `src/data` and `src/figures` though.

**Comment by jfcrenshaw** [comment]
Oh I also made some changes to the `Snakefile` and `showyourwork.yml` shown above. This was because when training models, I wanted to save the trained model along with the training losses, so those rules are now set to create directories instead of files, since it looks like showyourwork can only cache multi-output rules using a directory.

**Comment by rodluger** [comment]
Hmm. I don't have any ideas off the top of my head. I'll clone your repo tonight or tomorrow morning and play around with it to see if I can figure out what the issue is. Can you freeze the latest version of the cache so I have access to it?

**Comment by jfcrenshaw** [comment]
Ooh that revealed a clue! When I ran `showyourwork cache freeze`, I got the response `Zenodo Sandbox deposit not found for branch unknown.`

**Comment by rodluger** [comment]
Strange. Can you run [this command](https://github.com/showyourwork/showyourwork/blob/123ae9f1e1cd153b2a66cca2f4c1c0e559221fc7/showyourwork/git.py#L49) (``git branch --show-current``) in your terminal from the top level of your repo and tell me what happens?

**Comment by rodluger** [comment]
A few other tips:
- You can manually "freeze" the cache by going to Zenodo Sandbox in your browser and publishing the current draft of the deposit. Try that, and see if it helps?
- Try force-deleting the ``.showyourwork`` folder, which caches a bunch of stuff. Sounds like ``showyourwork clean`` failed to delete that. This could help resolve the issue once you re-run your workflow locally.
- I'll push some changes to ``showyourwork`` to make it print some more informative messages about why the cache was skipped. Stay tuned...

**Comment by jfcrenshaw** [comment]
Okay so when I ran `git branch --show-current`, git complained that `--show-current` wasn't an option. So I updated git  (I used `conda install git` since I don't have sudo access on this machine), and now it shows `main`. So I retried freezing again and got
```
❯ showyourwork cache freeze
Publishing Zenodo Sandbox deposit with concept DOI 10.5072/zenodo.1066333...
Request error 400: Validation error. Minimum one file must be provided.
```
so I think maybe it was some issue with the old git, and that showyourwork is able to see the zenodo cache again. I will try cleaning and re-running again to see if everything gets cached this time.

**Comment by jfcrenshaw** [comment]
By the way, is there a way to tell showyourwork to cache files that already exist? I haven't changed any inputs or scripts, so it would be nice if I could tell showyourwork to cache all the relevant files without having to rerun the whole workflow again.

**Comment by rodluger** [comment]
OK great! Good to know about git version issues -- will update the FAQs on the docs.

Keep me posted. In the meantime, one more tip I thought of:

The build logs can be useful. These are uploaded as a [build artifact](https://github.com/jfcrenshaw/pzflow-paper/suites/6688619487/artifacts/253815806). If you unzip it and look at ``logs/showyourwork.log``, you'll see the following:

```
Attempting to access Zenodo Sandbox deposit with DOI 10.5072/zenodo.1065931...
>>> Restoring from Zenodo Sandbox cache: src/data/two_moons_flow.pzflow.pkl...
Generating figure output: src/tex/figures/twomoons_example.pdf...
Running user rule train_main_galaxy_flow...
Searching remote file cache: src/data/main_galaxy_flow.pzflow.pkl...
Attempting to access Zenodo Sandbox deposit with DOI 10.5072/zenodo.1065931...
>>> File train_main_galaxy_flow found, but it has the wrong hash. Skipping...
File train_main_galaxy_flow not found in deposit with DOI 10.5072/zenodo.1065931.
Attempting to access Zenodo Sandbox record with DOI 10.5072/zenodo.1065931...
Required version of file not found in cache: src/data/main_galaxy_flow.pzflow.pkl.
Cache for src/data/main_galaxy_flow.pzflow.pkl not found, and `run_cache_rules_on_ci` is set to `False`.
```

So for the run you originally raised this issue for, it _did_ find the ``train_main_galaxy_flow`` file, but it had the wrong hash (the CI run had some modification to either the rule or the script or some upstream dependency).

**Comment by rodluger** [comment]
> By the way, is there a way to tell showyourwork to cache files that already exist? I haven't changed any inputs or scripts, so it would be nice if I could tell showyourwork to cache all the relevant files without having to rerun the whole workflow again.

I think it [already does what you're saying](https://github.com/showyourwork/showyourwork/blob/main/showyourwork/patches.py#L186-L207), but only once you've run ``showyourwork``. So run the workflow and _then_ freeze the cache -- it should work. Let me know if it doesn't.

**Comment by jfcrenshaw** [comment]
Again I got 
```
❯ showyourwork cache freeze
Publishing Zenodo Sandbox deposit with concept DOI 10.5072/zenodo.1066333...
Request error 400: Validation error. Minimum one file must be provided.
```
and when I go to the webpage for the zenodo sandbox deposit, there's no files listed.

Unless you have any other ideas, I'll go ahead and run `showyourwork clean`, make sure that `.showyourwork` gets deleted, and then rerun the workflow. Hopefully that resolves the issue

**Comment by rodluger** [comment]
OK, yes, do that. Keep me posted!

**Comment by jfcrenshaw** [comment]
Unfortunately it still didn't work 😔. 

I checked the log files under `.showyourwork/logs/` and I don't see any mentions of the cache. Could you add more info about the caching steps to the log so that I can get a better idea which stage is failing?

**Comment by rodluger** [comment]
Ok I'll play around with it today. And just to confirm, you set the SANDBOX_TOKEN secret on GitHub, right?

**Comment by jfcrenshaw** [comment]
Yes, but it's not cacheing on my personal computer either. When I look at the Zenodo sandbox webpage, it's empty. I know my SANDBOX_TOKEN is working because it was able to create the cache. I just don't know why it's not actually adding files to the cache

**Comment by rodluger** [comment]
OK, I can reproduce your original error on my fork.

Output when I run your workflow from scratch on my local computer (I switched to a new Sandbox DOI under my account, but no other changes to your repo):

```
Setting up the workflow...
Creating conda environment environment.yml...
Downloading and installing remote packages.
Environment for environment.yml created (location: .snakemake/conda/353aa0931262cc49df83b42aeafb820b)
Running user rule train_conditional_galaxy_flow...
Searching remote file cache: src/data/conditional_galaxy_flow...
Required version of file not found in cache: src/data/conditional_galaxy_flow.
Running rule from scratch...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Caching output file on remote: src/data/conditional_galaxy_flow...
################################################################################## 100.0%
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Running user rule train_two_moons_flow...
Searching remote file cache: src/data/two_moons_flow.pzflow.pkl...
Required version of file not found in cache: src/data/two_moons_flow.pzflow.pkl.
Running rule from scratch...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Caching output file on remote: src/data/two_moons_flow.pzflow.pkl...
################################################################################## 100.0%
Generating figure output: src/tex/figures/twomoons_example.pdf...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Running user rule train_main_galaxy_flow...
Searching remote file cache: src/data/main_galaxy_flow...
Required version of file not found in cache: src/data/main_galaxy_flow.
Running rule from scratch...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Caching output file on remote: src/data/main_galaxy_flow...
######################################################################################################################## 100.0%
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Generating the article PDF...
Done!
```

It successfully uploads all three cache files:

<img width="1165" alt="image" src="https://user-images.githubusercontent.com/9323819/171219673-23f4fae9-4f60-43f6-ac97-e7cfbbc48b3b.png">

But when I run the action on CI, I [get the same error](https://github.com/rodluger/pzflow-paper/runs/6674497236?check_suite_focus=true#step:3:351) saying it can't find the file in the cache:

```
Error: exec: Creating a new conda environment in ~/.showyourwork/env...
Setting up the workflow...
Creating conda environment environment.yml...
Downloading and installing remote packages.
Environment for environment.yml created (location: .snakemake/conda/46628ae2b6593b13269b1c16ba1db28c)
Running user rule train_conditional_galaxy_flow...
Searching remote file cache: src/data/conditional_galaxy_flow...
Required version of file not found in cache: src/data/conditional_galaxy_flow.
Cache for src/data/conditional_galaxy_flow not found, and `run_cache_rules_on_ci` is set to `False`.
```

So I can confirm this is an issue with ``showyourwork``. Will continue to investigate this morning.

As for the other issue where it's not caching on your local computer, something must have changed between when you raised the issue and now, since it looks like it was caching at first?

**Comment by jfcrenshaw** [comment]
Yeah it was caching before... I'll keep trying to figure that out

**Comment by jfcrenshaw** [comment]
Okay so the caching works from my laptop, but not from the department machine I like to run ML code on.

When running `showyourwork` on the dept machine, it never even prints dialog about caching, so that's not even being triggered for some reason.

Both machines are using `showyourwork 0.3.0.dev6`

On the dept machine:
```
❯ showyourwork
Running user rule train_conditional_galaxy_flow...
Running user rule train_main_galaxy_flow...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Running user rule train_two_moons_flow...
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating the article PDF...
Done!
```

versus on my laptop:
```
❯ showyourwork
Running user rule train_main_galaxy_flow...
Searching remote file cache: src/data/main_galaxy_flow...
Required version of file not found in cache: src/data/main_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/main_galaxy_flow...
######################################################################### 100.0%
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Running user rule train_conditional_galaxy_flow...
Searching remote file cache: src/data/conditional_galaxy_flow...
Required version of file not found in cache: src/data/conditional_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/conditional_galaxy_flow...
######################################################################### 100.0%
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Running user rule train_two_moons_flow...
Searching remote file cache: src/data/two_moons_flow.pzflow.pkl...
Required version of file not found in cache: src/data/two_moons_flow.pzflow.pkl.
Running rule from scratch...
Caching output file on remote: src/data/two_moons_flow.pzflow.pkl...
######################################################################### 100.0%
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating the article PDF...
Done!
```

**Comment by rodluger** [comment]
Interesting. I need another day or so to figure this out. In the meantime, your workflow is fast enough that you could set `run_cache_rules_on_ci: true` to get the action to pass.

**Comment by rodluger** [comment]
@jfcrenshaw I spent a while poking around and I take back what I said earlier -- when I fork your repository and create a new Zenodo Sandbox cache under my API token, things actually work as expected:

https://github.com/rodluger/pzflow-paper/runs/6697628686?check_suite_focus=true#step:3:116

Everything is properly restored from the cache and the article builds without a hitch. Earlier, when I said I was getting the same error, I had forgotten to create a SANDBOX_TOKEN GitHub Action secret, so the authentication failed resulting in a cache miss.

You can check the diff between the tip of my fork and the tip of your main branch here:

https://github.com/jfcrenshaw/pzflow-paper/compare/main...rodluger:main

If you scroll down to the changed files section, you'll see I didn't actually change anything substantial. I upgraded showyourwork to `v0.3.0.dev8` (more on this below), I added `src/data/cosmoDC2_subset.pkl` as an explicit dependency of  the scripts that load it, and I uploaded that file to a Zenodo Sandbox deposit with DOI 10.5072/zenodo.1068003 (see my comment in #100). I also changed the DOI in `zenodo.yml` to one I have API access to.

In terms of what changed on the ``showyourwork`` side, in `v0.3.0.dev8` I now added a bunch of debugging log messages to help the user debug issues with the cache. Most of these show up only in the build logs (``.showyourwork/logs/showyourwork.log``). On GitHub Actions, you can download the ``.showyourwork`` folder as an artifact from any workflow run (e.g., [here](https://github.com/rodluger/pzflow-paper/suites/6754236921/artifacts/258125487)) to inspect these logs. I also added informational log messages (with higher priority, so these show up in the terminal) telling the user whether their current `ZENODO_TOKEN` or `SANDBOX_TOKEN` has access to a given deposit on Zenodo. For my latest run, you can see that on [this line](https://github.com/rodluger/pzflow-paper/runs/6697628686?check_suite_focus=true#step:3:118) it says

```
User authentication for 10.5072/zenodo.1068003 is valid.
```

This should help users figure out if they've forgotten to set the GH Actions secret or if they've provided a wrong or invalid value.

So, back to your code: I suspect you either forgot to set the `SANDBOX_TOKEN` secret or you provided the wrong value. Try setting it again, and make sure the token has the correct scopes (`deposit:actions` and `deposit:write`). Either way, if you upgrade to `v0.3.0.dev8`, it should tell you whether the token has access to the cache.

I didn't create a PR for you because you should use a Sandbox deposit for which you have API access. So take a look at the diff I linked to above, upload `dc2.pkl` to a deposit you have write access to, and edit the relevant lines in the `showyourwork.yml` file. If things fail again, download the build artifact and take a look at the logs. If things are _still_ failing, you can always mimic the behavior on CI by running

```
CI=true showyourwork
```

(i.e., set the `CI=true` environment variable and run ``showyourwork``). Run ``showyourwork clean`` for a fresh build.

As for why the workflow isn't even trying to access the cache on your dept machine, I don't know, but hopefully the extra level of debug logging might help.

Keep me posted!

**Comment by jfcrenshaw** [comment]
@rodluger first of all, thanks so much for all of your help! It's really impressive how quick and responsive you are!

On my dept machine, I started with a clean repo and clean install. I copied over the relevant changes from your diff, and upgraded to `v0.3.0.dev8`. Specifically, `showyourwork --version` shows `0.3.0.dev8` and `showyourwork.yml` has `version: 0.3.0.dev8` at the top.

I then created a new Zenodo sandbox cache using `showyourwork cache create`. This was successful, and I confirmed online that the cache exists.

I then ran `showyourwork`, and got the following output:
```
❯ showyourwork
Setting up the workflow...
Creating conda environment environment.yml...
Downloading and installing remote packages.
Environment for environment.yml created (location: .snakemake/conda/8a6b2681ea58482fdf16284367bba157)
Running user rule train_main_galaxy_flow...
Running user rule train_conditional_galaxy_flow...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Running user rule train_two_moons_flow...
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Generating the article PDF...
Done!
```

Notice there is no mention of caching. There is also no Zenodo debugging info in the log. My `.showyourwork/logs/showyourwork.log` shows

```
Setting up the workflow...
Creating conda environment environment.yml...
Downloading and installing remote packages.

... skipping all the conda install stuff ...

Environment for environment.yml created (location: .snakemake/conda/8a6b2681ea58482fdf16284367bba157)
Running user rule train_main_galaxy_flow...
Running user rule train_conditional_galaxy_flow...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Running user rule train_two_moons_flow...
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Generating the article PDF...
Done!
```

I was worried if somehow the update to `v0.3.0.dev8` didn't propagate. But I checked `.showyourwork/showyourwork.yml` and confirmed it shows `version: 0.3.0.dev8`. I also checked `.showyourwork/config.json` which also shows `"version": "0.3.0.dev8"`.

I then pushed the changes to Github and cloned everything onto my personal laptop. I also copied the exact same `SANDBOX_TOKEN` from my dept machine to my laptop.

I ran `showyourwork`, and got the following output
```
❯ showyourwork
Setting up the workflow...
User authentication for 10.5072/zenodo.1069391 is valid.
Creating conda environment environment.yml...
Downloading and installing remote packages.
Environment for environment.yml created (location: .snakemake/conda/e8f99b34214df6ee91c51c4fae9c4a5a)
Running user rule train_main_galaxy_flow...
Searching remote file cache: src/data/main_galaxy_flow...
Required version of file not found in cache: src/data/main_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/main_galaxy_flow...
############################################################################## 100.0%
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Running user rule train_conditional_galaxy_flow...
Searching remote file cache: src/data/conditional_galaxy_flow...
Required version of file not found in cache: src/data/conditional_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/conditional_galaxy_flow...
############################################################################## 100.0%
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Running user rule train_two_moons_flow...
Searching remote file cache: src/data/two_moons_flow.pzflow.pkl...
Required version of file not found in cache: src/data/two_moons_flow.pzflow.pkl.
Running rule from scratch...
Caching output file on remote: src/data/two_moons_flow.pzflow.pkl...
############################################################################## 100.0%
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating the article PDF...
Done!
```

In addition, `.showyourwork/logs/showyourwork.log` shows
```
Setting up the workflow...
Testing if user is authenticated for 10.5072/zenodo.1069391...
User authentication for 10.5072/zenodo.1069391 is valid.
Creating conda environment environment.yml...
Downloading and installing remote packages.

... skipping all the conda install stuff ...

Environment for environment.yml created (location: .snakemake/conda/e8f99b34214df6ee91c51c4fae9c4a5a)
Running user rule train_main_galaxy_flow...
Searching remote file cache: src/data/main_galaxy_flow...
Searching Zenodo cache...
Zenodo DOI not provided in `zenodo.yml`.
Searching Zenodo Sandbox cache...
Attempting to access Zenodo Sandbox deposit with DOI 10.5072/zenodo.1069391...
Searching for file `train_main_galaxy_flow` with hash `7f1ce31b53917ddb563f88eea0987c407979d0da5ac934ae58764e57f47c753d`...
File train_main_galaxy_flow not found in deposit with DOI 10.5072/zenodo.1069391.
Attempting to access Zenodo Sandbox record with DOI 10.5072/zenodo.1069391...
Required version of file not found in cache: src/data/main_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/main_galaxy_flow...
Generating figure output: src/tex/figures/main_galaxy_corner.pdf...
Running user rule train_conditional_galaxy_flow...
Searching remote file cache: src/data/conditional_galaxy_flow...
Searching Zenodo cache...
Zenodo DOI not provided in `zenodo.yml`.
Searching Zenodo Sandbox cache...
Attempting to access Zenodo Sandbox deposit with DOI 10.5072/zenodo.1069391...
Searching for file `train_conditional_galaxy_flow` with hash `461d9ed4208616aa6b4d21aedb8e0222fa6abae5a4d0bf55a3e7e60627637ae0`...
Inspecting candidate file `train_main_galaxy_flow` with hash `461d9ed4208616aa6b4d21aedb8e0222fa6abae5a4d0bf55a3e7e60627637ae0`...
Cache miss for file {entry['filename']}. Skipping...
File train_conditional_galaxy_flow not found in deposit with DOI 10.5072/zenodo.1069391.
Attempting to access Zenodo Sandbox record with DOI 10.5072/zenodo.1069391...
Required version of file not found in cache: src/data/conditional_galaxy_flow.
Running rule from scratch...
Caching output file on remote: src/data/conditional_galaxy_flow...
Generating figure output: src/tex/figures/conditional_galaxy_corner.pdf...
Generating figure output: src/tex/figures/galaxy_flow_losses.pdf...
Running user rule train_two_moons_flow...
Searching remote file cache: src/data/two_moons_flow.pzflow.pkl...
Searching Zenodo cache...
Zenodo DOI not provided in `zenodo.yml`.
Searching Zenodo Sandbox cache...
Attempting to access Zenodo Sandbox deposit with DOI 10.5072/zenodo.1069391...
Searching for file `train_two_moons_flow` with hash `14ff808c26600817362e70af2ffa6c5931e6dc0ef2dbb03d00b774114f26d738`...
Inspecting candidate file `train_conditional_galaxy_flow` with hash `14ff808c26600817362e70af2ffa6c5931e6dc0ef2dbb03d00b774114f26d738`...
Cache miss for file {entry['filename']}. Skipping...
Inspecting candidate file `train_main_galaxy_flow` with hash `14ff808c26600817362e70af2ffa6c5931e6dc0ef2dbb03d00b774114f26d738`...
Cache miss for file {entry['filename']}. Skipping...
File train_two_moons_flow not found in deposit with DOI 10.5072/zenodo.1069391.
Attempting to access Zenodo Sandbox record with DOI 10.5072/zenodo.1069391...
Required version of file not found in cache: src/data/two_moons_flow.pzflow.pkl.
Running rule from scratch...
Caching output file on remote: src/data/two_moons_flow.pzflow.pkl...
Generating figure output: src/tex/figures/twomoons_example.pdf...
Generating the article PDF...
Done!
```

As you can see, when I run the exact same code on my laptop, the caching and new caching debugging dialog works fine. I looked at the cache online and confirmed all the files were there. 

Furthermore, after `showyourwork` on my laptop caches everything on Zenodo, the Github action runs perfectly. So the only problem is with running `showyourwork` on my dept machine. I know this is the hardest thing for you to debug since you don't have access to that machine! (unless you still have UW credentials?)

But somehow, `showyourwork` is interacting with the computing environment in some mysterious way that is changing its behavior on the two machines. It's weird that not only is the caching not occurring on the dept machine, but none of the caching dialog gets printed as well.

I'll keep thinking about this, but just wanted to give you an update, and see if you had any ideas.

Also, if you want, I can close this issue and open a new one for the dept machine caching not working, since the github caching is working fine now.

**Comment by rodluger** [comment]
Ok, glad we were able to isolate the problem! Yes, please do copy this over to a separate issue (and close this one), mostly to benefit others who might run into the same issue(s). I'll poke around the code a bit to see if I can think of anything.

But right off the bat, this smells like another ``git`` version issue. Can you check that you're running ``git>=2.22.0``?
